src/services/api.ts

```diff
@@ -189,11 +189,18 @@ const sendEmailToDoctor = async (token: string, prescriptionId: number, careDeta
   return response.data
 }
 
-const getSubscriptionById = async (token: string, id: number): Promise<Subscription> => {
+const getSubscriptionById = async (token: string, id: number): Promise<Subscription | null> => {
   const url = API_V1 + `/payment/subscription/${id}/`
   const headers = { Authorization: `Token ${token}` }
-  const response = await axios.get<Subscription>(url, { headers })
-  return response.data
+  try {
+    const response = await axios.get<Subscription>(url, { headers });
+    return response.data;
+  } catch (error) {
+    if (axios.isAxiosError(error) && error.response?.status === 404) {
+      return null;
+    }
+    throw error;
+  }
```

Comment:
What led to these changes?

Reply:
when I get a subscription associated with the logged-in user if he doesn't have a subscription I get `{“detail”: “No StripeProduct matches the given query.”}`

Reply:
I see, yeah that makes sense thanks.
Why not handling the exception later in the stack since you're going to have a branching to test the null case anyway?
So I mean in `src/context/subscription.tsx` instead of:
```
if (subscription === null ) return setSubscription(defaultSubscriptionData);
```
Maybe do the try/catch there, that would make the overall code slightly simpler.
Also add a comment in the try/catch to explain just what you said regarding the no subscription.

Reply:
I get it